driver/src/main/java/com/code_intelligence/jazzer/junit/RegressionTestExtensions.java

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.code_intelligence.jazzer.junit;
+
+import com.code_intelligence.jazzer.agent.AgentInstaller;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext;
+
+class RegressionTestExtensions
+    implements BeforeEachCallback, AfterEachCallback, InvocationInterceptor {
+  private static Field lastFindingField;
+
+  @Override
+  public void beforeEach(ExtensionContext extensionContext) {
```

Comment:
Wouldn't it make sense to use `BeforeAll` as these functions can only be executed effectively ones?

Reply:
I tried that before and found it not to work. I think it's because we would have to specify the `BeforeAllCallback` on the class rather than the method, but we don't control that.